### ISSUE TITLE
match typescript-extended rules inside `rules` block - `buildFromOxlintConfig`

### ISSUE THREAD

**Issue by Sysix** [oxc-project/eslint-plugin-oxlint#228]
Some of the typescript-extended rules are inherently supported by `oxlint` and thus are only advertised as a single rule.  
It's a bit annoying having to match rules when they are different names.

_Originally posted by @drvn-mr in https://github.com/oxc-project/eslint-plugin-oxlint/issues/226#issuecomment-2457554398_

**Comment by drvn-mr** [comment]
I should've provided an example but I have a script parsing the `oxlint --rules` output to match our rules.
This comes out for rules that are in both `eslint` and `@typescript-eslint`, for which only a single rule exists in `oxlint`:
```
{
  "@typescript-eslint/default-param-last": "default-param-last",
  "@typescript-eslint/no-array-constructor": "no-array-constructor",
  "@typescript-eslint/no-empty-function": "no-empty-function",
  "@typescript-eslint/no-unused-vars": "no-unused-vars",
  "@typescript-eslint/no-useless-constructor": "no-useless-constructor"
}
```

**Comment by Sysix** [comment]
oh your use case is fixed with #231. 

I found another use case which is not supported:

```
  it('disables matching typescript and eslint rules', () => {
    for (const rule in [
      'no-unused-vars',
      '@typescript-eslint/no-unused-vars',
    ]) {
      const rules = buildFromOxlintConfig({
        rules: {
          [rule]: 'warn',
        },
      });

      expect(rules.length).toBe(1);
      expect(rules[0].rules).not.toBeUndefined();
      expect('no-unused-vars' in rules[0].rules!).toBe(true);
      expect('@typescript-eslint/no-unused-vars' in rules[0].rules!).toBe(true);
    }
  });

```

**Comment by Sysix** [comment]
After revisiting the Problem and fixing it for `vitest` and `jest` rules, 
I do not think my use case is really helpful.

Closing this issue :)
